File: fibonacci.rb
def sum(n)
  if n<2
    return 1
  end
  return n+sum(n-1)

end

def fib(n)
  if n<2
    return 0
  end
  if n<3
    return 1
  end

  return fib(n-2)+fib(n-1)
end

def main
  start = Time.now
  puts fib(42)
  finish = Time.now
  puts "Time = #{finish-start}"
end

main

File: fibonacci.go
package main

import (
  "fmt"
  "time"
)

func main() {
  defer timeTrack(time.Now(),"Fibonacci")

  for i := 0; i < 100; i++ {
    fmt.Printf("fib %d %d \n" ,i, fib(42))
  }
}

func fib(n int)int  {
  if n < 2{
    return 0
  }
  if n < 3{
    return 1
  }
  return fib(n-1)+fib(n-2)
}

func timeTrack(start time.Time, name string) {
    elapsed := time.Since(start)
    fmt.Printf("%s took %s", name, elapsed)
}
